metrics/web/docs/features/sentry/01_sentry_design.md

```diff
@@ -0,0 +1,167 @@
+- an `initialize` method , that simplify [initializing the Sentry](#initializing-the-sentry) with an ability to [capture uncaught Dart and Flutter errors](#capturing-uncaught-flutter-and-dart-errors), set the DSN and release name;
+- a `captureException` method, that delegates to the `Sentry` method described in the [Capturing exceptions](#capturing-exceptions);
+- an `addContext` method, that delegates to the `Sentry` method described in the [Adding Contexts](#adding-contexts).
+
+The following class diagram represents the classes required for Sentry integration: 
+
+![Sentry class diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/platform-platform/monorepo/raw/sentry_design/metrics/web/docs/features/sentry/diagrams/sentry_util_class_diagram.puml)
+
+## Data Source Name (DSN)
+
+The [DSN](https://docs.sentry.io/product/sentry-basics/dsn-explainer/) tells the SDK where to send the events. If this value is not provided, the SDK will try to read it from the `SENTRY_DSN` environment variable. If that variable also does not exist, the SDK will just not send any events.
+
+The `DSN` is not a secret, worst thing someone could do is sending events to your account. If that ever happens we have a few options tackling this, we can either [block off certain request](https://docs.sentry.io/product/accounts/quotas/#inbound-data-filters) or cycle the `DSN`.
+
+Despite the above, we still should to improve our security in using the `DSN`, so we should build the Flutter app providing the `DNS` enviroment variable. Also, the environment variable save us from the hard-coding the `DSN`.
+
+```bash
+export SENTRY_DSN=`your_dsn_here`
+flutter build web --dart-define=SENTRY_DSN=$SENTRY_DSN
+```
+
+Then in the app code we can take the `DSN` using `String.fromEnviroment()` method:
+
+```dart
+const dsn = const String.fromEnvironment('SENTRY_DSN');
+```
+
+## Updating source maps
+
+Due to flutter minifying, the release app errors are not readable in the Sentry. To resolve the problem we should [update the js sourcemaps](https://docs.sentry.io/platforms/javascript/sourcemaps). 
+
+To get it work in the `Dart` we should build the Flutter app providing the `release name` and updating the source maps:
+
+```bash
+export SENTRY_RELEASE=release-name
+flutter build web --dart-define=SENTRY_RELEASE=$SENTRY_RELEASE
+
+# Upload source maps
+sentry-cli releases new $SENTRY_RELEASE
+
+sentry-cli releases files $SENTRY_RELEASE upload-sourcemaps . \
+    --ext dart \
+    --rewrite
+
+cd ./build/web/
+sentry-cli releases files $SENTRY_RELEASE upload-sourcemaps . \
+    --ext map \
+    --ext js \
+    --rewrite
+
+sentry-cli releases finalize $SENTRY_RELEASE
+```
+
+Then in the app code, we should take the `release name` using `String.fromEnviroment()` method and pass it to the `SentryUtil.init` method:
+
+```dart
+const release = const String.fromEnvironment('SENTRY_RELEASE');
+```
+
```

Comment:
```suggestion

## Testing

```
I suppose we should add information on how we can test Sentry logging. Consider the following points to cover: 
- note the sentry has the `.bindClient` method (https://pub.dev/documentation/sentry/latest/sentry/Sentry/bindClient.html);
- note that using this method we can use the mockito to test sentry;
- note that as sentry is a third party API it possible should be tested using [third party API testing](https://github.com/platform-platform/monorepo/blob/master/docs/03_third_party_api_testing.md) with [mock server](https://github.com/platform-platform/monorepo/blob/master/docs/04_mock_server.md).

Not sure we will test Sentry integration widely but we have to describe the possible testing.